common.sh

```diff
@@ -354,16 +359,33 @@ fi
 
 # Agent specific configuration 
 
+function invalidAgentValue() {
+  printf "Found invalid value \"$AGENT_E2E_TEST_SCENARIO\" for AGENT_E2E_TEST_SCENARIO. \nSupported values: \nCOMPACT_IPV4\nCOMPACT_IPV6\nDHCP_COMPACT_IPV4\nDHCP_COMPACT_IPV6\nHA_IPV4\nHA_IPV6\nDHCP_HA_IPV4\nDHCP_HA_IPV6\nSNO_IPV4\nSNO_IPV6\nDHCP_SNO_IPV4\nDHCP_SNO_IPV6\n"
+  exit 1
+}
+
 # Agent test scenario
 export AGENT_E2E_TEST_SCENARIO=${AGENT_E2E_TEST_SCENARIO:-}
+export NETWORKING_MODE=${NETWORKING_MODE:-}
 
 # Enable MCE deployment
 export AGENT_DEPLOY_MCE=${AGENT_DEPLOY_MCE:-}
 
 
 if [[ ! -z ${AGENT_E2E_TEST_SCENARIO} ]]; then
-  export IP_STACK=$(echo ${AGENT_E2E_TEST_SCENARIO##*_IP} | tr V v)
-  SCENARIO=${AGENT_E2E_TEST_SCENARIO%%_*}
+  IFS='_'
+  read -a arr <<<"$AGENT_E2E_TEST_SCENARIO"
+  delimiterCount=$(echo "$AGENT_E2E_TEST_SCENARIO" | tr -cd '_' | wc -c)
+
+  SCENARIO=${arr[0]}
+  export IP_STACK=${arr[1]}
```

Comment:
Accepted values for `IP_STACK` are `v4`, `v6` or `v4v6`, so it seems that `IP` must be stripped off from `arr[1]`, ie: ```export IP_STACK=${arr[1]##*IP}``` (in the previous implementation there was a similar string manipulation)